src/commcare_cloud/ansible/roles/bootstrap-users/tasks/main.yml

```diff
@@ -64,21 +64,22 @@
   tags:
     - new_ansible_sudo_pass
 
-- name: Authorize current devs to login as ansible
-  become: yes
-  authorized_key: user=ansible state=present key="{{ lookup('file', authorized_keys_dir ~ item ~ '.pub') }}"
+- name: remove key using username
+  lineinfile:
+    dest: /home/ansible/.ssh/authorized_keys
+    state: absent
+    regexp: "username={{ item }}"
+  when: reset_user | default(False)
   with_items: '{{ dev_users.present }}'
   tags:
     - pubkey
 
-# task is remove old pubkey of particular devs from ansible and update with new pubkey
-- name: UnAuthorize current devs to login as ansible
+- name: Authorize current devs to login as ansible
   become: yes
-  authorized_key: user=ansible state=absent key="{{ lookup('file', authorized_keys_dir ~ item ~ '.pub') }}"
-  with_items: '{{ user }}'
+  authorized_key: user=ansible state=present key="{{ lookup('file', authorized_keys_dir ~ item ~ '.pub') }}" key_options=environment="username={{ item }}"
```

Comment:
Will existing keys be updated automatically the first time this is run? If not, what is the plan to update existing authorized keys?

Reply:
we need to run the `update-users` in order to make this change available on all machines. Still thinking about the alternative procedure to make this option available. Please let me know your suggestions too. thanks